app/api/v1/endpoints/admin.py

```
@@ -0,0 +1,244 @@
+
+        log_user_action(
+            admin.email,
+            f"deleted user {user_email}"
+        )
+
+        return {
+            "message": "User deleted successfully",
+            "detail": f"User {user_email} and all related data have been permanently deleted"
+        }
+    except ValueError as e:
+        raise HTTPException(status_code=400, detail=str(e))
```

Ruff (0.14.5)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)